src/xr/xr-plane-detection.js

```diff
@@ -0,0 +1,138 @@
+import { EventHandler } from '../core/event-handler.js';
+import { XrPlane } from './xr-plane.js';
+
+/**
+ * @class
+ * @name XrPlaneDetection
+ * @classdesc Plane Detection provides the ability to detect real world surfaces based on estimations of underlying AR system.
+ * @description Plane Detection provides the ability to detect real world surfaces based on estimations of underlying AR system.
+ * @param {XrManager} manager - WebXR Manager.
+ * @property {boolean} supported True if Plane Detection is supported.
+ * @property {boolean} available True if Plane Detection is available. This property can be set to true only during running session.
+ * @property {XrPlane[]} planes List of {@link XrPlane} that contain individual plane information.
+ * @example
+ * // start session with plane detection enabled
+ * app.xr.start(camera, pc.XRTYPE_VR, pc.XRSPACE_LOCALFLOOR, {
+ *     planeDetection: true
+ * });
+ * @example
+ * app.xr.planeDetection.on('add', function (plane) {
+ *     // new plane been added
+ * });
+ */
+class XrPlaneDetection extends EventHandler {
+    constructor(manager) {
+        super();
+
+        this._manager = manager;
+        this._supported = !! window.XRPlane;
+        this._available = false;
+
+        this._planesIndex = new Map();
```

Comment:
it'd be great to have comment for the Map here, with info about what is the key and what is the value